test/components/tableTests.ts

```diff
@@ -1,370 +1,472 @@
+      assertTableRows(table, expectedRows, "removing a nonexistent Component does not affect the table");
+    });
 
-    spaceRequest = table.requestedSpace(90, 90);
-    TestMethods.verifySpaceRequest(spaceRequest, 70, 100, "3");
+    it("has no further effect when called a second time with the same Component", () => {
+      const tableRows = [
+        [c1, c2, c3],
+        [c4, c5, c6]
+      ];
+      table = new Plottable.Components.Table(tableRows);
+      const expectedRows = [
+        [null, c2, c3],
+        [c4, c5, c6]
+      ];
+      table.remove(c1);
+      assertTableRows(table, expectedRows, "Component was removed");
+      table.remove(c1);
+      assertTableRows(table, expectedRows, "removing Component again has no further effect");
+    });
 
-    spaceRequest = table.requestedSpace(200, 200);
-    TestMethods.verifySpaceRequest(spaceRequest, 70, 100, "4");
+    it("removes a Component from the Table if the Component becomes detached", () => {
+      table = new Plottable.Components.Table([[c1]]);
+      c1.detach();
+      assert.isNull(table.componentAt(0, 0), "calling detach() on the Component removes it from the Table");
+      assert.isNull(c1.parent(), "Component disconnected from the Table");
+    });
   });
 
-  describe("table._iterateLayout works properly", () => {
-    // This test battery would have caught #405
-    function verifyLayoutResult(result: any,
-                                cPS: number[], rPS: number[], gW: number[], gH: number[],
-                                wW: boolean, wH: boolean, id: string) {
-      assert.deepEqual(result.colProportionalSpace, cPS, "colProportionalSpace:" + id);
-      assert.deepEqual(result.rowProportionalSpace, rPS, "rowProportionalSpace:" + id);
-      assert.deepEqual(result.guaranteedWidths, gW, "guaranteedWidths:" + id);
-      assert.deepEqual(result.guaranteedHeights, gH, "guaranteedHeights:" + id);
-      assert.deepEqual(result.wantsWidth, wW, "wantsWidth:" + id);
-      assert.deepEqual(result.wantsHeight, wH, "wantsHeight:" + id);
-    }
+  describe("requesting space", () => {
+    it("is fixed-width if all columns contain only fixed-width Components, non-fixed otherwise", () => {
+      const fixedNullsTable = new Plottable.Components.Table([
+        [new Mocks.FixedSizeComponent(), null],
+        [null, new Mocks.FixedSizeComponent()]
+      ]);
+      assert.isTrue(fixedNullsTable.fixedWidth(), "width is fixed if all columns contain fixed-width Components or null");
 
-    it("iterateLayout works in the easy case where there is plenty of space and everything is satisfied on first go", () => {
-      let c1 = new Mocks.FixedSizeComponent(50, 50);
-      let c2 = new Plottable.Component();
-      let c3 = new Plottable.Component();
-      let c4 = new Mocks.FixedSizeComponent(20, 10);
-      let table = new Plottable.Components.Table([
-        [c1, c2],
-        [c3, c4]
+      const someUnfixedTable = new Plottable.Components.Table([
```

Comment:
Only some, but not all of the columns are unfixed. Should I name it `someUnfixedColumnsTable`?